src/components/fields/location-input-group/location-modal/location-search/location-search.tsx

```diff
@@ -0,0 +1,749 @@
+							x: parseFloat(res.results?.[0].X) || undefined,
+							y: parseFloat(res.results?.[0].Y) || undefined,
+						};
+						updateFormValues(fetchedFormValue);
+						changeSelectedAddressInfo(fetchedFormValue);
+						// populate result list
+						setQueryString(fetchedFormValue.address);
+						setUseQuerySearch(true);
+					} else {
+						updateFormValues({});
+						changeSelectedAddressInfo({});
+					}
+				} catch (error) {}
+			})();
+			// if only latlng
+		} else if (reverseGeoCodeEndpoint && !formValues?.address && formValues?.lat && formValues?.lng) {
+			(async () => {
+				try {
+					const locationList = await OneMapService.reverseGeocode({
+						route: reverseGeoCodeEndpoint,
+						latitude: formValues.lat,
+						longitude: formValues.lng,
+					});
+					const postalCodeCheck =
+						!mustHavePostalCode || LocationHelper.hasGotAddressValue(locationList?.[0].POSTALCODE);
+					if (locationList?.[0] && postalCodeCheck) {
+						const address = LocationHelper.formatAddressFromGeocodeInfo(locationList?.[0], true);
+						// Get full address
+						const fetchedFormValue = {
+							address,
+							blockNo: locationList[0].BLOCK,
+							building: locationList[0].BUILDINGNAME,
+							postalCode: locationList[0].POSTALCODE,
+							roadName: locationList[0].ROAD,
+							lat: formValues.lat,
+							lng: formValues.lng,
+							x: parseFloat(locationList[0].XCOORD) || undefined,
+							y: parseFloat(locationList[0].YCOORD) || undefined,
+						};
+						updateFormValues(fetchedFormValue);
+						changeSelectedAddressInfo(fetchedFormValue);
+						// populate result list
+						setQueryString(fetchedFormValue.address);
+						setUseQuerySearch(true);
+					}
+				} catch (error) {}
+			})();
+		}
+	}, []);
+
+	useEffect(() => {
+		if (!showLocationModal) return;
+
+		if (gettingCurrentLocation) {
+			getUserCurrentLocation(disableErrorPromptOnApp);
+			return;
+		}
+
+		if (!navigator.onLine) return;
+
+		// internet restored
+		if (!queryString && selectedAddressInfo?.lat && selectedAddressInfo?.lng) {
+			getAddressInfo(selectedAddressInfo.lat, selectedAddressInfo.lng);
+			return;
+		} else {
+			// check if one map or reverse code is working
+			// - get location error
+			// - first load
+			// the map should not be usable if any of these services fail
+			debounceFetchAddress("singapore", 1, undefined, onOneMapError);
+			const reverseGeocodeCheck = async () => {
+				try {
+					reverseGeocodeAborter.current?.abort();
+					reverseGeocodeAborter.current = new AbortController();
+					// esclipse latlng
+					await OneMapService.reverseGeocode({
+						route: reverseGeoCodeEndpoint,
+						latitude: 1.29994179707526,
+						longitude: 103.789404349716,
+						bufferRadius: 1,
+						abortSignal: reverseGeocodeAborter.current.signal,
+						otherFeatures: OneMapBoolean.YES,
+					});
+					reverseGeocodeAborter.current = null;
+				} catch (error) {
+					onOneMapError();
+				}
+			};
+			reverseGeocodeCheck();
```

Comment:
move to `helpers.ts`, can do `Promise.all`, do add comments to explain what it is meant to do